kolibri/core/device/utils.py

```diff
@@ -141,3 +111,219 @@ def set_app_key_on_response(response):
+    return new_settings
+
+
+def validate_device_settings(language_id=None, facility=None, **new_settings):
+    # Override any settings passed in
+    for key in new_settings:
+        check_device_setting(key)
+
+    settings_to_set = dict(new_settings)
+    if language_id is not None:
+        settings_to_set["language_id"] = language_id
+    if facility is not None:
+        settings_to_set["default_facility"] = facility
+
+    return settings_to_set
+
+
+def create_facility(facility_name=None, preset=None):
+    from kolibri.core.auth.models import Facility
+
+    facility = Facility.objects.create(name=facility_name)
+    logger.info("Facility with name '{name}' created.".format(name=facility.name))
+
+    # Only set preset data if we have created the facility, otherwise leave previous data intact
+    if preset:
+        dataset_data = mappings[preset]
+        facility.dataset.preset = preset
+        for key, value in dataset_data.items():
+            setattr(facility.dataset, key, value)
+        facility.dataset.save()
+        logger.info("Facility preset changed to {preset}.".format(preset=preset))
+    return facility
+
+
+def setup_device_and_facility(
+    facility,
+    facility_name,
+    preset,
+    facility_settings,
+    device_settings,
+    username,
+    password,
+):
+    from kolibri.core.auth.models import FacilityUser
+
+    with transaction.atomic():
+        if facility is None and facility_name is not None:
+            facility = create_facility(
+                facility_name=facility_name,
+                preset=preset,
+            )
+
+            if facility_settings:
+                for key, value in facility_settings.items():
+                    setattr(facility.dataset, key, value)
+                facility.dataset.save()
+                logger.info(
+                    "Facility settings updated with {}".format(facility_settings)
+                )
+
+        provision_device(**device_settings)
+        logger.info("Device settings updated with {}".format(device_settings))
+
+        if username and password and facility:
+            try:
+                FacilityUser.objects.create_superuser(
+                    username, password, facility=facility
+                )
+                logger.info(
+                    "Superuser created with username '{username}' in facility '{facility}'.".format(
+                        username=username, facility=facility
+                    )
+                )
+            except ValidationError:
+                logger.warn(
+                    "An account with username '{username}' already exists in facility '{facility}', not creating user account.".format(
+                        username=username, facility=facility
+                    )
+                )
+
+
+def provision_from_file(file_path):
+    """
+    Expects a JSON file with the following format (example values supplied):
+    {
+        "facility": "My Facility",
```

Comment:
```suggestion
        "facility_name": "My Facility",
```

otherwise might be confused with facility ID

Reply:
Yeah, that was my instinct also, I had only not changed it because of a holdover from the provisiondevice management command.